### PR TITLE
Don't dispatch select event if no feature is selected or deselected

### LINKED ISSUE #5422: ol.interaction.Select: too many select event
`select` event are dispatched even if no feature is selected or deselected.

To reproduce the issue (?):
 - http://openlayers.org/en/v3.16.0/examples/select-features.html
 - set a beakpoint at https://github.com/openlayers/ol3/blob/v3.16.0/examples/select-features.js#L74
 - mouse click in the ocean

The `select`event is dispatched but no feature is selected or deselected.

Is it the expected behavior ?

### PR DESCRIPTION
**Pull request by fredj**
Fixes #5422

### CODE REVIEW

**Comment by fredj** [comment]
TODO: with this change, a `select` event is dispatched if an already selected feature is selected again.

**Comment by fredj** [comment]
> TODO: with this change, a `select` event is dispatched if an already selected feature is selected again.

fixed

**Comment by marcjansen** [comment]
Do you still consider this `[wip]`?

**Comment by fredj** [comment]
It's now ready for review, thanks

File: src/ol/interaction/selectinteraction.js [hunk 1]
```diff
@@ -282,11 +281,10 @@ ol.interaction.Select.handleEvent = function(mapBrowserEvent) {
             return !this.multi_;
           }
         }, this, this.layerFilter_);
-    if (selected.length > 0 && features.getLength() == 1 &&
-        features.item(0) == selected[0]) {
-      // No change
+    if (selected.length > 0 && features.getLength() == 1 && features.item(0) == selected[0]) {
+      // No change; an already selected feature is selected again
+      selected.length = 0;
```
Comment (fredj) [inline src/ol/interaction/selectinteraction.js]: fixme: check this

**Comment by fredj** [comment]
@ahocevar can you please take a look at this pull request? (no hurry ...)

**Comment by ahocevar** [comment]
Looks good. But like every change in `ol.interaction.Select` - fingers crossed. Please merge.

**Pull request merged**